.circleci/config.yml

```diff
@@ -1,109 +1,303 @@
       - run:
-          name: Build New Strings .pot
+          name: Lint Client and Server
+          when: always
           command: |
-            if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              git clone https://github.com/Automattic/gp-localci-client.git
-              bash gp-localci-client/generate-new-strings-pot.sh $CIRCLE_BRANCH $CIRCLE_SHA1 $CIRCLE_ARTIFACTS/translate
-              rm -rf gp-localci-client
+            # We'll use failure code to do some additional processing and report failure manually
+            set +o errexit
+            ./node_modules/.bin/eslint-eslines                        \
+              $(                                                      \
+                git diff --name-only --diff-filter=d origin/master... \
+                | grep -E '^(client|server)'                          \
+                | grep -E '\.jsx?$'                                   \
+              )                                                       \
+              -- --format junit                                       \
+              > ~/eslint-results.xml
+            # Ensure we don't upload empty reports.
+            # Eslint would not produce a malformed empty xml using the --format and -o options.
+            # However, with eslint-eslines and the redirected > output, we may get an empty file.
+            # We only put the file into the reports directory if the status code informs us that
+            # lint issues have been encountered.
+            ESLINT_RESULT="$?"
+            if [ "$ESLINT_RESULT" -ne 0 ]; then
+              mv ~/eslint-results.xml "$CIRCLE_TEST_REPORTS/eslint/results.xml"
+              exit "$ESLINT_RESULT"
             fi
-
       - run:
-          name: Run Integration Tests
+          name: Build calypso-strings.pot
+          when: always
           command: |
-            bin/run-integration $(circleci tests glob "bin/**/integration/*.js" "client/**/integration/*.js" "server/**/integration/*.js" | circleci tests split --split-by=timings)
-
-      - run:
-          name: Lint Config Keys
-          command: npm run lint:config-defaults
-
+            npm run translate
+            mv calypso-strings.pot "$CIRCLE_ARTIFACTS/translate"
       - run:
-          name: Lint Client and Server
+          name: Build New Strings .pot
+          when: always
           command: |
-            ./node_modules/.bin/eslint-eslines $(git diff --name-only --diff-filter=d origin/master... | grep -E '^(client|server)' | grep -E '\.jsx?$' | circleci tests split)
+            git clone https://github.com/Automattic/gp-localci-client.git
+            bash gp-localci-client/generate-new-strings-pot.sh "$CIRCLE_BRANCH" "$CIRCLE_SHA1" "$CIRCLE_ARTIFACTS/translate"
+            rm -rf gp-localci-client
+      - store_test_results:
+          path: /tmp/test_results
+      - store_artifacts:
+          path: /tmp/test_results
+      - store_artifacts:
+          path: /tmp/artifacts
```

Comment:
Why are there two `store_artifacts` (with different `path`s) specified?

Reply:
This uploads different things associated with the CI job. We can likely rework this so that both directories can be uploaded in a single step. The artifacts contain output from translation. Uploading the test results just makes the whole test report available for someone to access at a later point. This existed before and hasn't really changed with this PR.